tt_metal/hw/inc/dataflow_api.h

```diff
@@ -579,7 +579,7 @@ struct InterleavedAddrGenFast {
         }
 
         DEBUG_STATUS('N', 'R', 'T', 'W');
-        DEBUG_SANITIZE_NOC_ADDR(NOC_XY_ADDR2(src_noc_xy, src_addr), this->page_size);
+        DEBUG_SANITIZE_NOC_ADDR((uint64_t)src_noc_xy | (uint64_t)src_addr, this->page_size);
```

Comment:
This is missing the bit shift of 32 for noc_xy.

We could use the helper func `get_noc_addr_helper(src_noc_xy, src_addr)`

Reply:
yeah, duh, thanks!